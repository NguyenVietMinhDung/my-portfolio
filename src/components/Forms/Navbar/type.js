import type { Props as Items } from './components/NavbarMenu/type';

export type Props = {
  items: Items,
  logoUrl: string,
  hamburgerIconUrl: string,
  closeIconUrl: string,
  activatedIndex: number,
  isVisible: boolean,
  activateNavigationItem: Function,
  openNavigationMenu: Function,
};
