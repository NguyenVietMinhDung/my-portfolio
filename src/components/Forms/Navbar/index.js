import Navbarbar from './Navbar';

export default Navbarbar;
