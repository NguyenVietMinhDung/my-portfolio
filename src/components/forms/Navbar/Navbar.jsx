// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './Navbar.scss';
import type { Props } from './type';
import { Brand } from '../../commons';
import NavbarMenu from './components/NavbarMenu';
import HamburgerButton from './components/HamburgerButton';

const Navbar = (props: Props) => {
  const {
    items,
    logoUrl,
    hamburgerIconUrl,
    closeIconUrl,
    activatedIndex,
    isVisible,
    activateNavigationItem,
    openNavigationMenu,
  } = props;
  return (
    <div styleName="nav">
      <div styleName="content">
        <Brand logoUrl={logoUrl} />
        <NavbarMenu
          isVisible={isVisible}
          items={items}
          activatedIndex={activatedIndex}
          activateNavigationItem={activateNavigationItem}
          openNavigationMenu={openNavigationMenu}
        />
        <HamburgerButton
          isVisible={isVisible}
          closeIconUrl={closeIconUrl}
          hamburgerIconUrl={hamburgerIconUrl}
          openNavigationMenu={openNavigationMenu}
        />
      </div>
    </div>
  );
};

export default memo<Props>(CSSModules(Navbar, styles));
