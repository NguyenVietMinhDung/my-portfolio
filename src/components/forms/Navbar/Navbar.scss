@import '../../../styles/abstracts/variables';

.nav {
	position: fixed;
	z-index: 2;

	display: flex;

	width: 100%;
	padding: 30px 0;

	background-color: $white-color;
	box-shadow: 0 0 30px 0 $shadow-color;
	flex-flow: row wrap;
	justify-content: center;

	.content {
		position: relative;

		display: flex;

		width: $layout-breakpoint;
		padding: 0 15px;
		align-items: center;
		flex-flow: row wrap;
		justify-content: space-between;
	}
}
