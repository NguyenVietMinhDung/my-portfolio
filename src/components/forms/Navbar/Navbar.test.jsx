import React from 'react';
import { shallow } from 'enzyme';
import props from '../../../constants/navbar';
import Navbar from './Navbar';

const defaultProps = {
  ...props,
  activatedIndex: 0,
  isVisible: false,
  activateNavigationItem: jest.fn(),
  openNavigationMenu: jest.fn(),
};

describe('<Navbar />', () => {
  test('renders correctly', () => {
    const component = shallow(<Navbar {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
