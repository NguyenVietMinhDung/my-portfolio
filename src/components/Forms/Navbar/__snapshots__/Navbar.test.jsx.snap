// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<div
  className="nav"
>
  <div
    className="content"
  >
    <Memo(Brand) />
    <Memo(NavbarMenu)
      activateNavigationItem={[MockFunction]}
      activatedIndex={0}
      isVisible={false}
      openNavigationMenu={[MockFunction]}
    />
    <Memo(HamburgerButton)
      isVisible={false}
      openNavigationMenu={[MockFunction]}
    />
  </div>
</div>
`;
